index.js

```diff
@@ -65,6 +65,10 @@ module.exports = {
             require('./lib/s3middleware/validateConditionalHeaders')
             .validateConditionalHeaders,
         MD5Sum: require('./lib/s3middleware/MD5Sum'),
+        azureMpuUtils: require('./lib/s3middleware/azureMpuUtils'),
```

Comment:
for the azureMpuUtils, ResultsCollector and SubStreamInterface (and maybe objectUtils but they seem pretty generic) I would nest them under an azure subheading so they are separated since these only apply to putting to azure.

Reply:
I ended up not needing the ResultsCollector and SubStreamInterface in S3, so have removed them. Is it worth putting just the azureMpuUtils in it's own directory?